src/coverlet.core/Coverage.cs

```diff
@@ -160,13 +162,20 @@ public CoverageResult GetCoverageResult()
             return coverageResult;
         }
 
-        private void CalculateCoverage()
+        private void CalculateCoverage(bool excludeNonCalledFiles)
         {
+            var resultsToRemove = new List<InstrumenterResult>();
             foreach (var result in _results)
             {
                 if (!File.Exists(result.HitsFilePath))
                 {
-                    // File not instrumented, or nothing in it called.  Warn about this?
+                    // File not instrumented, or nothing in it called.
+                    // Mark to be removed so no non used modules are included in coverage file?
+                    if (excludeNonCalledFiles)
+                    {
+                        resultsToRemove.Add(result);
```

Comment:
@MarcoRossignoli this made the magic.
I just included a parameter for not doing this by default, because @tonerdo wanted to keep original behaviour (for some reason)
But this solved excluding any specific stuff that's not part of your code. Together with loop in line `245`

Reply:
At the moment I don't have a strong opinion, but I think that if we support this feature should be throught switch(like in this PR) because could be also useful understand why some module are not "tested", in big project someone could lose "the problem" of "disabled" tests.

Reply:
Not sure if i can think an scenario in which you wouldn't see not tested modules which are part of your code.
Maybe if you don't test Api lvl ,then api.dll won't pop up and coverage will be higher. That's what you mean? but then you immediately don't see it covered in your coverage report or console output

Reply:
> but then you immediately don't see it covered in your coverage report or console output

Yes but I think a scenario where I have tens of modules...if I see module with 0% it's immediatly clear that something goes wrong(for instance temporary commented tests and not re-enabled) if you trim that result I could lose the issue, make sense?